archive/backup.go

```diff
@@ -25,32 +28,58 @@ func CreateBackup(
 	from uint64,
 	to *uint64,
 	outPath string,
+	overwriteFile bool,
+	enableZstdCompression bool,
+	zstdLevel int,
 ) (uint64, uint64, error) {
-	// always create new file, throw error if the file exists
-	fs, err := os.OpenFile(outPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0644)
+	// allow to overwrite the overwrites file only if it's explicitly set
+	fileFlag := os.O_WRONLY | os.O_CREATE | os.O_EXCL
+	if overwriteFile {
+		fileFlag = os.O_WRONLY | os.O_CREATE | os.O_TRUNC
+	}
+
+	fp, err := os.OpenFile(outPath, fileFlag, 0644)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	closeFile := func() error {
-		if err := fs.Close(); err != nil {
-			logger.Error("an error occurred while closing file", "err", err)
-
-			return err
+	defer func() {
+		err := fp.Close()
+		if err != nil {
+			logger.Error("Failed to close the file", "err", err)
 		}
+	}()
 
-		return nil
-	}
-	removeFile := func() {
-		if err := os.Remove(outPath); err != nil {
-			logger.Error("an error occurred while removing file", "err", err)
+	fbuf := bufio.NewWriterSize(fp, 1*1024*1024)
+
+	defer func() {
```

Comment:
Should we use only one `defer` here for more readable?
By the way, `error` should be returned when failed.